test/common/testlib.py

```diff
@@ -1305,6 +1307,23 @@ def write_file(self, path, content):
         if self.is_nondestructive():
             self.addCleanup(m.execute, "rm -f {0}".format(path))
 
+    def check_coverage(self):
+        """Collect code coverage result and save to file
+
+        Hash result as part of file name, like "coverage-<coverage hash result>.json",
+        to avoid duplicate coverage file and save file to ".nyc_output" folder.
+        """
+        coverage_data = self.browser.eval_js("window.__coverage__", no_trace=True)
```

Comment:
`window.__coverage__` does not exist in firefox nor chrome.
Where does this come from? Is it some plugin in your browser? If so, then tests need to be able to install/load this plugin into browser.

Reply:
This's based on UI test code coverage tool [istanbul](https://istanbul.js.org/), which needs a babel plugin [babel-plugin-istanbul](https://github.com/istanbuljs/babel-plugin-istanbul). The plugin [instruments code](https://github.com/weldr/cockpit-composer/blob/7f295a8ed5da2dd91f3f0328f480dc6beeb3e72c/webpack.config.js#L19) with Istanbul coverage. c-composer did this [here](https://github.com/weldr/cockpit-composer/blob/7f295a8ed5da2dd91f3f0328f480dc6beeb3e72c/Makefile#L9).
And cockpit ci container has the file (~/.config/codecov-token) already.